packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -257,6 +259,68 @@ export default function createInstantSearchManager({
     }
   }
 
+  function hydrateSearchClient(client, results) {
+    if (!results) {
+      return;
+    }
+
+    if (!client._useCache) {
+      // This condition avoid hydrating a `searchClient` different from the
+      // Algolia one. Unless the implementation uses the exact same approach
+      // for the cache (the property `_useCache`). The implementation is brittle
+      // but we don't have a proper way to detect the Algolia client at the moment.
+      return;
+    }
+
+    if (Array.isArray(results)) {
+      hydrateSearchClientWithMultiIndexRequest(client, results);
+      return;
```

Comment:
In that case we could use `if/else` because it's be easy to miss some instructions since the multi-index path ends here.

Reply:
It  depends a lot on your habits, I don't have a strong opinion about that `if/else` is equivalent.

Reply:
I meant that the early return is quite misleading here since you don't necessarily want to end the path here, you rather don't want to call `hydrateSearchClientWithSingleIndexRequest`.

Reply:
Actually I do want to stop the path at this point.